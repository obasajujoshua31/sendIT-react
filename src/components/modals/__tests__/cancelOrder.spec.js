import React from "react";
import { shallow } from "enzyme";
import sinon from "sinon";
import {
  mapDispatchToProps,
  mapStateToProps,
  CancelParcelOrder
} from "../cancelOrder";

const props = {
  parcelId: "1",
  role: false,
  history: {
    push: jest.fn()
  },
  show: true,
  status: "PLACED",
  closeCancel: false,
  cancelParcelOrder: jest.fn(),
  destination: "dkkdkd",
  parcelStatus: false
};
const mockStore = {
  parcels: {
    parcelStatus: true
  }
};
const mockSubmit = {
  preventDefault: jest.fn()
};


const wrapper = shallow(<CancelParcelOrder {...props} />);

describe("Edit CancelParcelOrder Test", () => {
  it("should render Edit CancelParcelOrder", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("dispatch should call loadOneParcel", () => {
    const dispatchSpy = sinon.spy();
    const dispatchResult = mapDispatchToProps(dispatchSpy);
    dispatchResult.cancelParcelOrder("1");
    expect(dispatchSpy.callCount).toEqual(1);
  });

  it("map state to props should return parcels", () => {
    const mapStateResult = mapStateToProps(mockStore);
    expect(mapStateResult.parcelStatus).toEqual(true);
  });

  it("cancelParcel should call props.cancelParcelStatus", () => {
    const cancelParcelSpy = jest.spyOn(wrapper.instance(), "cancelParcel");
    cancelParcelSpy(mockSubmit);
    expect(wrapper.instance().props.cancelParcelOrder).toBeCalledWith("1");
  });

 
  it("should call props.history.push when parcelStatus is true", () => {
      wrapper.setProps({
          parcelStatus: true
      });
      expect(wrapper.instance().props.history.push).toBeCalledWith(
          "/dashboard"
      );
  });

  it("should render status modal if state is the same as props", () => {
      wrapper.setState({
          parcelStatus: false
      });
      wrapper.setProps({
          parcelStatus: false
      });
      expect(wrapper).toMatchSnapshot();
  });
});
