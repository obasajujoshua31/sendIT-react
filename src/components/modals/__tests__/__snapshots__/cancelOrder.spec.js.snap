// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edit CancelParcelOrder Test should render Edit CancelParcelOrder 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="editModal"
  closeTimeoutMS={0}
  contentLabel="Modal #1 Global Style Override Example"
  isOpen={true}
  onRequestClose={false}
  overlayClassName="Overlay"
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <div
    style={
      Object {
        "alignItem": "center",
        "backgroundColor": "#806B75",
        "display": "flex",
        "justifyContent": "center",
        "padding": 15,
        "position": "relative",
      }
    }
  >
    <span
      style={
        Object {
          "bottom": 5,
          "color": "white",
          "cursor": "pointer",
          "fontSize": "1.8em",
          "padding": 5,
          "position": "absolute",
          "right": 5,
        }
      }
    >
      <i
        className="far fa-times-circle"
        onClick={false}
      />
    </span>
    <h3
      style={
        Object {
          "color": "white",
          "fontSize": "1.4rem",
          "fontWeight": "bold",
        }
      }
    >
      Form Cancellation Confirmation
    </h3>
  </div>
  <div
    style={
      Object {
        "alignItem": "center",
        "display": "flex",
        "justifyContent": "center",
        "maxWidth": "100%",
        "padding": "10px",
      }
    }
  >
    <form
      noValidate={true}
      onSubmit={[Function]}
    >
      <button
        style={
          Object {
            "backgroundColor": "#fd9b01",
            "border": "none",
            "color": "white",
            "margin": 0,
            "outline": "none",
            "padding": "10px 20px",
            "position": "relative",
          }
        }
      >
        Confirm Cancel Order
      </button>
    </form>
  </div>
</Modal>
`;

exports[`Edit CancelParcelOrder Test should render status modal if state is the same as props 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="editModal"
  closeTimeoutMS={0}
  contentLabel="Modal #1 Global Style Override Example"
  isOpen={true}
  onRequestClose={false}
  overlayClassName="Overlay"
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <div
    style={
      Object {
        "alignItem": "center",
        "backgroundColor": "#806B75",
        "display": "flex",
        "justifyContent": "center",
        "padding": 15,
        "position": "relative",
      }
    }
  >
    <span
      style={
        Object {
          "bottom": 5,
          "color": "white",
          "cursor": "pointer",
          "fontSize": "1.8em",
          "padding": 5,
          "position": "absolute",
          "right": 5,
        }
      }
    >
      <i
        className="far fa-times-circle"
        onClick={false}
      />
    </span>
    <h3
      style={
        Object {
          "color": "white",
          "fontSize": "1.4rem",
          "fontWeight": "bold",
        }
      }
    >
      Form Cancellation Confirmation
    </h3>
  </div>
  <div
    style={
      Object {
        "alignItem": "center",
        "display": "flex",
        "justifyContent": "center",
        "maxWidth": "100%",
        "padding": "10px",
      }
    }
  >
    <form
      noValidate={true}
      onSubmit={
        [MockFunction] {
          "calls": Array [
            Array [
              Object {
                "preventDefault": [MockFunction] {
                  "calls": Array [
                    Array [],
                  ],
                  "results": Array [
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                  ],
                },
              },
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        }
      }
    >
      <button
        style={
          Object {
            "backgroundColor": "#fd9b01",
            "border": "none",
            "color": "white",
            "margin": 0,
            "outline": "none",
            "padding": "10px 20px",
            "position": "relative",
          }
        }
      >
        Confirm Cancel Order
      </button>
    </form>
  </div>
</Modal>
`;
